src/daemon/distribution.h

```diff
@@ -0,0 +1,37 @@
+#ifndef DISTRIBUTION_H
+#define DISTRIBUTION_H
+
+#include "collectd.h"
+
+struct distribution_s;
+typedef struct distribution_s distribution_t;
+
+/* function that create new distribution structure and initialize buckets using
+ * linear function */
+distribution_t *distribution_new_linear(size_t num_buckets, double size);
+/* function that create new distribution structure and initialize buckets using
+ * exponential function */
+distribution_t *distribution_new_exponential(size_t num_buckets,
+                                             double initial_size,
+                                             double factor);
+/* function that create new distribution structure and initialize buckets using
+ * custom buckets sizes given by the user */
+distribution_t *distribution_new_custom(size_t num_buckets,
+                                        double *custom_buckets_sizes);
+
+/* function for updating the buckets */
+void distribution_update(distribution_t *d, double gauge);
+
+/* function  for getting the percentile */
```

Comment:
I really appreciate that all the functions in the header file are commented, that's awesome :)

I recommend that the comments discuss the function's behavior in corner cases. For example, what constraints exist for the arguments, such as «"percent" has to be between 0 (exclusive) and 100 (inclusive)». Also, how are error conditions signaled? Will the function return zero on failure? Or -1? Or NaN (Not a Number)? Is `errno` set?

Reply:
Thank you! I extended the comments of descriptions of values returned in case of any error